src/Microsoft.AspNet.Mvc.Razor/RazorTextWriter.cs

```diff
@@ -99,15 +99,7 @@ public override void Write(string value)
         /// <inheritdoc />
         public override void Write(IHtmlContent value)
         {
-            var htmlTextWriter = TargetWriter as HtmlTextWriter;
-            if (htmlTextWriter == null)
-            {
-                value.WriteTo(TargetWriter, HtmlEncoder);
-            }
-            else
-            {
-                htmlTextWriter.Write(value);
-            }
+            value.WriteTo(TargetWriter, HtmlEncoder);
```

Comment:
Could you add this as a note so it doesn't get optimized by accident in the future?